exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/sqlite/ForeignKeyConstraintTests.kt

```diff
@@ -71,4 +68,120 @@ class ForeignKeyConstraintTests : DatabaseTestsBase() {
             Item.select { Item.id eq tabboulehId }.single()[Item.categoryId]
         )
     }
+
+    @Test
+    fun `test ON DELETE RESTRICT for databases that support it without SQLite`() {
+        withDb(excludeSettings = listOf(TestDB.SQLITE, TestDB.SQLSERVER)) {
+            testOnDeleteRestrict()
+        }
+    }
+
+    @Test
+    fun `test ON DELETE RESTRICT for SQLite`() {
+        Assume.assumeTrue(TestDB.SQLITE in TestDB.enabledDialects())
+
+        transaction(Database.connect("jdbc:sqlite:file:test?mode=memory&cache=shared&foreign_keys=on", user = "root", driver = "org.sqlite.JDBC")) {
+            testOnDeleteRestrict()
+        }
+    }
+
+    private fun testOnDeleteRestrict() {
+        val country = object : Table("Country") {
+            val id = integer("id")
+            val name = varchar(name = "name", length = 20)
+
+            override val primaryKey = PrimaryKey(id)
+        }
+
+        val city = object : Table("City") {
+            val id = integer("id")
+            val name = varchar(name = "name", length = 20)
+            val countryId = integer("countryId")
+                .references(
+                    country.id,
+                    onDelete = ReferenceOption.RESTRICT
+                )
+
+            override val primaryKey = PrimaryKey(id)
+        }
+
+        SchemaUtils.drop(country, city)
+        SchemaUtils.create(country, city)
+
+        val lebanonId = 0
+        country.insert {
+            it[id] = lebanonId
+            it[name] = "Lebanon"
+        }
+
+        val beirutId = 0
+        city.insert {
+            it[id] = beirutId
+            it[name] = "Beirut"
+            it[countryId] = 0
+        }
+
+        expectException<ExposedSQLException> {
+            country.deleteWhere { id eq lebanonId }
+        }
+    }
+
+    @Test
+    fun `test ON UPDATE RESTRICT for databases that support it without SQLite`() {
+        withDb(excludeSettings = listOf(TestDB.SQLITE, TestDB.SQLSERVER, TestDB.ORACLE)) {
+            testOnUpdateRestrict()
+        }
+    }
+
+    @Test
+    fun `test ON UPDATE RESTRICT for SQLite`() {
+        Assume.assumeTrue(TestDB.SQLITE in TestDB.enabledDialects())
+
+        transaction(Database.connect("jdbc:sqlite:file:test?mode=memory&cache=shared&foreign_keys=on", user = "root", driver = "org.sqlite.JDBC")) {
+            testOnUpdateRestrict()
+        }
+    }
+
+    private fun testOnUpdateRestrict() {
```

Comment:
Since everything is the same except the `expectException` block, could the duplicate code be put in a helper function?
Or maybe it gets reduced to one function `testRestrict()` with a toggle parameter for `onUpdate` or `onDelete` that affects what happens in the former block?
Or at minimum the 2 tables could be reused by the functions?